src/libraries/System.Text.Json/tests/Serialization/Stream.Collections.cs

```diff
@@ -353,6 +353,16 @@ private static IEnumerable<Type> DictionaryTypes<TElement>()
             typeof(GenericIReadOnlyDictionaryWrapper<string, TElement>)
         };
 
+        // Non-generic types cannot roundtrip when they contain a $ref written on serialization and they are the root type.
```

Comment:
What's a minimal code sample to highlight this behavior?

Reply:
key1 will be parsed as a `JsonElement` and when key2 tries to look for the reference, it will not find it.
```cs
using System;
using System.Collections;
using System.Text.Json;
using System.Text.Json.Serialization;

public class Program
{
  public static void Main()
  {
    var foo = new Foo();
    Hashtable h = new Hashtable();
    h.Add("key1", foo);
    h.Add("key2", foo);

    var opts = new JsonSerializerOptions();
    opts.ReferenceHandler = ReferenceHandler.Preserve;

    string json = JsonSerializer.Serialize(h, opts);
    Console.WriteLine(json);
    // {"$id":"1","key1":{"$id":"2"},"key2":{"$ref":"2"}}

    Hashtable copyOfh = JsonSerializer.Deserialize<Hashtable>(json, opts);
  }
}
```